### PR TITLE
feat: added AWS SNS service

### PR DESCRIPTION
**Pull request by Prajval-Aggarwal**
### Description
Created a service of AWS SNS

### Changes Made
- Added a new service of simple text messaging using the AWS SNS service.

### Related Issues
NIL

### Testing Done
Testing is done from the backend side

### CODE REVIEW

File: server/handler/alert_services/aws_sns.go
Comment (AmanTarar) [inline server/handler/alert_services/aws_sns.go:1]: error handling pending in Request Decoding function.

**Review by AmanTarar — COMMENTED** [review]
Remove unnecessary logs and do error handling wherever required.

File: server/handler/alert_services/aws_sns.go
Comment (Prajval-Aggarwal) [reply server/handler/alert_services/aws_sns.go:1]: Done

File: server/handler/alert_services/aws_sns.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"main/server/request"
+	"main/server/response"
+
+	awssns "main/server/services/alert_service/aws_sns"
+	"main/server/utils"
+
+	"github.com/gin-gonic/gin"
+)
+
+func AwsTextMessaging(ctx *gin.Context) {
```
Comment (SalmaanChicmic) [inline server/handler/alert_services/aws_sns.go:13]: More often than not, this service will be used by other services/clients, for boilerplate we can code it that way

File: server/services/alert_service/aws_sns/text_messaging.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/gin-gonic/gin"
+)
+
+func SendSMS(ctx *gin.Context, reqBody request.AwsTextMessagingRequest) {
+
+	// Create Session and assign AccessKeyID and SecretAccessKey
+	sess, err := session.NewSession(&aws.Config{
```
Comment (SalmaanChicmic) [inline server/services/alert_service/aws_sns/text_messaging.go:19]: Will we create a new session each time this method is called?

File: server/services/alert_service/aws_sns/text_messaging.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+	// sends a text message (SMS message) directly to a phone number.
+	resp, err := svc.Publish(params)
+
+	if err != nil {
+		response.ShowResponse(err.Error(), utils.HTTP_BAD_REQUEST, utils.FAILURE, nil, ctx)
+		return
+	}
+
+	// fmt.Println(resp) // print the response data.
```
Comment (SalmaanChicmic) [inline server/services/alert_service/aws_sns/text_messaging.go:48]: please remove unused comments

File: server/services/alert_service/aws_sns/text_messaging.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		Credentials: credentials.NewStaticCredentials(os.Getenv("AWS_ACESSS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), ""),
+	},
+	)
+	if err != nil {
+		response.ShowResponse(err.Error(), utils.HTTP_INTERNAL_SERVER_ERROR, utils.FAILURE, nil, ctx)
+		return
+	}
+
+	// Create SNS service
+	svc := sns.New(sess)
```
Comment (SalmaanChicmic) [inline server/services/alert_service/aws_sns/text_messaging.go:30]: what is svc?

File: server/handler/alert_services/aws_sns.go (hunk 1 above)
Comment (Prajval-Aggarwal) [reply server/handler/alert_services/aws_sns.go:13]: Sir, can you please explain a bit more?

File: server/services/alert_service/aws_sns/text_messaging.go (hunk 2 above)
Comment (Prajval-Aggarwal) [reply server/services/alert_service/aws_sns/text_messaging.go:19]: Resolved sir. Created another function which creates a session and assigned it to global variable .

File: server/services/alert_service/aws_sns/text_messaging.go (hunk 3 above)
Comment (Prajval-Aggarwal) [reply server/services/alert_service/aws_sns/text_messaging.go:48]: Removed..

File: server/services/alert_service/aws_sns/text_messaging.go (hunk 4 above)
Comment (Prajval-Aggarwal) [reply server/services/alert_service/aws_sns/text_messaging.go:30]: svc is the instance of sns with a session.

File: server/handler/alert_services/aws_sns.go (hunk 1 above)
Comment (SalmaanChicmic) [reply server/handler/alert_services/aws_sns.go:13]: Right now this is being invoked by an api call right? But usually these are called via other controllers/services. It's fine you can leave it as is.

File: server/services/alert_service/aws_sns/text_messaging.go (hunk 4 above)
Comment (SalmaanChicmic) [reply server/services/alert_service/aws_sns/text_messaging.go:30]: What does `svc` variable name stand for?

**Review by SalmaanChicmic — APPROVED** [review]

File: server/handler/alert_services/aws_sns.go (hunk 1 above)
Comment (Prajval-Aggarwal) [reply server/handler/alert_services/aws_sns.go:13]: Okay sir

File: server/services/alert_service/aws_sns/text_messaging.go (hunk 4 above)
Comment (Prajval-Aggarwal) [reply server/services/alert_service/aws_sns/text_messaging.go:30]: Sir,It stands for service. It is common name for variable when using aws services.

**Pull request closed without merging**